jmeter-java-dsl/src/main/java/us/abstracta/jmeter/javadsl/codegeneration/MethodCall.java

```diff
@@ -339,29 +360,18 @@ public void prependChild(MethodCall child) {
    *                                       found to be chained in current method call.
    */
   public MethodCall chain(String methodName, MethodParam... params) {
-    // this eases chaining don't having to check in client code for this condition
-    if (params.length > 0 && Arrays.stream(params).allMatch(MethodParam::isDefault)) {
+    if (shouldSkipChaining(params)) {
       return this;
     }
-    /*
-    when chaining methods with booleans in some cases the parameter is required, and in some others
-    is not.
-     */
-    Method method = null;
-    if (params.length == 1 && params[0] instanceof BoolParam) {
-      method = findMethodInClassHierarchyMatchingParams(methodName, returnType, new MethodParam[0]);
-      if (method != null) {
-        params = new MethodParam[0];
-      }
-    }
-    if (method == null) {
-      method = findMethodInClassHierarchyMatchingParams(methodName, returnType, params);
-    }
+
+    MethodParam[] effectiveParams = adjustParamsForBoolParam(methodName, params);
```

Comment:
I like the way of putting the logic for booleans in a specific method instead of having a comment. But, this refactor requires searching for the method twice, which is not much performance penalty, and in general I think is better to have better readability/maintainability and gaining such small performance improvement. What about something like:

```java
public MethodCall chain(String methodName, MethodParam... params) {
  if (params.length > 0 && Arrays.stream(params).allMatch(MethodParam::isDefault)) {
    return this;
  }
  chain.add(buildMethodCall(method, params));
  return this;
}

public MethodCall buildMethodCall(String methodName, MethodParam... params) {
  if (params.length == 1 && params[0] instanceof BoolParam) {
    Method ret = findMethodInClassHierarchyMatchingParams(methodName, returnType, new MethodParam[0]);
    if (ret != null) {
      return MethodCall.from(ret, new MethodParam[0]);
    }
  }
  Method ret = findMethodInClassHierarchyMatchingParams(methodName, returnType, params);
  if (ret == null) {
    throw buildNoMatchingMethodFoundException("public '" + methodName + "' method in " + returnType.getName(), params);
  }
  return MethodCall.from(ret, params);
}
```